src/unix.rs

```diff
@@ -435,6 +436,14 @@ unsafe fn fd_check(fd: c_int, check_pipe: bool) -> Result<(), FromEnvErrorInner>
     }
 }
 
+fn clone_fd_and_set_cloexec(fd: c_int) -> Result<File, FromEnvErrorInner> {
```

Comment:
You can use `BorrowedFd::try_clone_to_owned` and then `File`'s `From<OwnedFd>` here. Not sure if it's shorter but probably more clear safety-wise.

Reply:
Thanks, but this PR is already merged.

I suggest opening a new PR for this.

Reply:
Sorry for the confusing comment (which I have deleted), I will apply your suggestion once I'm free.

Reply:
@carbotaniuman BorrowedFd is introdiced in 1.63, but cc (which uses this crate) has msrv set to 1.46 https://github.com/rust-lang/cc-rs/blob/360754b91441395713c24db6e5a5e3a91b298527/.github/workflows/main.yml#L124

Unfortunately I can't apply your suggestion due to this.

Reply:
Looks like jobserver has already used some new API and was bumped to 1.63 a while back, so I guess we could use that?

Reply:
Thanks I didn't notice that, will apply your suggestion.

Reply:
Updated.